### PR TITLE
update push CI workflow files for security

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

See [internal discussion](https://huggingface.slack.com/archives/C01NE71C4F7/p1724750779061389?thread_ts=1724157181.387969&cid=C01NE71C4F7)

### CODE REVIEW

File: .github/workflows/self-push-amd.yml [hunk 1]
```diff
@@ -76,11 +81,7 @@ jobs:
         # `CI_SHA_WORKFLOW_RUN`: The commit SHA that triggers this workflow by `workflow_run` event
         # `CI_SHA`: The non-empty commit SHA from the above two (one and only one of them is empty)
         run: |
-          CI_BRANCH_PUSH=${{ github.event.ref }}
```
Comment (amyeroberts) [inline .github/workflows/self-push-amd.yml:79]: We should move the corresponding comments about the different CI values to the new place

**Review by amyeroberts — APPROVED** [review]
Thanks for handling!

File: .github/workflows/self-push-amd.yml (hunk 1 above)
Comment (ydshieh) [reply .github/workflows/self-push-amd.yml:79]: Thanks, will do before merge

**Review by LysandreJik — APPROVED** [review]
SG!

**Pull request merged**